src/components/ChatEntry.js

```diff
@@ -1,22 +1,66 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+import { useState } from 'react';
 
 const ChatEntry = (props) => {
+  //const timeStampObj = new Date(props.timeStamp);
+  //const now = new Date();
+  //const year = now.getFullYear() - timeStampObj.getFullYear();
+
+  const [heart, setHeart] = useState('🤍');
+
+  const changeHeart = () => {
+    if (heart === '🤍') {
+      setHeart('❤️');
+      props.onHeartChange(true);
+    } else {
+      setHeart('🤍');
+      props.onHeartChange(false);
+    }
+  };
```

Comment:
This all works well, but state is not fully lifted, and the message data at the App level is out of sync with the content here at ChatEntry.  If you wanted to lift state up out of this component into app, onHeartChange would need to pass the id of this ChatEntry up to App.  The chat message data from the JSON file would need to be saved in a state variable, and then the callback function would need to update that state variable appropriately.  Here in ChatEntry, the 'liked' property from the message data could then be used to select the correct heart to display, and no state variables would be needed here.